spring-cloud-commons/src/main/java/org/springframework/cloud/client/serviceregistry/AbstractAutoServiceRegistration.java

```diff
@@ -84,20 +84,21 @@ protected ApplicationContext getContext() {
 
 	@Override
 	@SuppressWarnings("deprecation")
-	public void onApplicationEvent(WebServerInitializedEvent event) {
+	public void onApplicationEvent(ApplicationReadyEvent event) {
 		bind(event);
 	}
 
 	@Deprecated
-	public void bind(WebServerInitializedEvent event) {
+	public void bind(ApplicationReadyEvent event) {
 		ApplicationContext context = event.getApplicationContext();
 		if (context instanceof ConfigurableWebServerApplicationContext) {
 			if ("management".equals(((ConfigurableWebServerApplicationContext) context)
 					.getServerNamespace())) {
 				return;
 			}
 		}
-		this.port.compareAndSet(0, event.getWebServer().getPort());
+		String port = context.getEnvironment().getProperty("server.port");
```

Comment:
This will not work of it had not been set

Reply:
> This will not work of it had not been set

Enmm, I did not consider the reason that server.port is not set, I will change it. The reason why I am modifying the source code is because in some scenarios, the user sets the web container himself, but I hope that springcloud can automatically register the service if it listens. Is the WebServerInitializedEvent event, it is actually unable to automatically register the service.